pkg/api/customization/secret/validator.go

```
@@ -0,0 +1,58 @@
+package secret
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/rancher/norman/api/access"
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+	v1 "github.com/rancher/types/apis/core/v1"
+)
+
+func Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
+	if request.Method == http.MethodPost {
+		id := ""
+
+		// extracting project name from data
+		if projectData, ok := data["projectId"].(string); ok {
+			if projectParts := strings.Split(projectData, ":"); len(projectParts) > 1 {
```

Should we return an error if by some reason len(projectParts) <=1?